vllm/model_executor/guided_decoding/xgrammar_decoding.py

```diff
@@ -152,7 +155,12 @@ def from_guided_params(cls,
                        tokenizer_hash=tokenizer_hash,
                        max_threads=max_threads)
         elif guided_params.grammar:
-            return cls(grammar_str=guided_params.grammar,
+            # XGrammar only supports GBNF grammars, so we must convert Lark
+            if grammar_is_likely_lark(guided_params.grammar):
+                grammar_str = convert_lark_to_gbnf(guided_params.grammar)
```

Comment:
do a try catch here we can raise better error to the user something like "Failed to convert the grammar from Lark to GBNF, please use GBNF grammar or specify --guided-decoding-backend=outlines"

Reply:
Yup, this is a good call